src/nvim/api/vim.c

```diff
@@ -695,6 +695,56 @@ Dictionary nvim_get_color_map(void)
   return colors;
 }
 
+/// Get the mappings for the corresponding scope and mode
+///
+/// @param mode
+/// @param scope
+Array nvim_get_keymap(String mode, String scope)
+  FUNC_API_SINCE(2)
+{
+  Array mappings = ARRAY_DICT_INIT;
+  mapblock_T *current_maphash;
+
+  // Convert the string mode to the integer mode
+  // that is stored within each mapblock
+  int int_mode = get_map_mode((char_u **)(&(mode.data)), 0);
+  int buffer_local = 0;
+
+  if (strcmp(scope.data, "buffer") == 0) {
+    buffer_local = 1;
+  }
+
+  int i = 0;
+  for (i = 0; i < MAX_MAPHASH; i++) {
+    // Get the correct maphash based on global or local hashes
+    current_maphash = get_maphash(i, scope.data);
+
+    if (current_maphash == NULL) {
+      continue;
+    }
+
+    while (current_maphash) {
```

Comment:
Should not it be

```C
for (mapblock_T *current_maphash = get_maphash(i, scope.data); current_maphash; current_maphash = current_maphash->m_next)
```

? And you definitely need no `if()` with `continue` at all, even with your code: while cycle just will not be executed. Also `current_maphash` variable declaration must be in the cycle, not at the start of the function.